### PR TITLE
Ft  association classes

### PR DESCRIPTION
**Pull request by Bateyjosue**
# FEATURE DESCRIPTION
## Task:
In this project, I'm going to finish creating the remaining classes for our school library and create the associations between them.

1. Create a class `Classroom` 

- [ ] Create the has-many/belongs-to relationship between `Classroom` and `Student`.
- [ ] Create the has-many side (a classroom has many students).
- [ ] Create the belongs-to side (a student belongs to a classroom).

2. Create a class `Book` 
3. Create a class `Rental` 

- [ ] Create the has-many side of `Book` and `Rental` (a book has many rentals).
- [ ] Create the belongs-to side of `Rental` and `Book` (a rental belongs to a book).
- [ ] Create the has-many side of `Person` and `Rental` (a person has many rentals).
- [ ] Create the belongs-to side of `Rental` and `Person` (a rental belongs to a person).

## Cheers

# Happy Review

### CODE REVIEW

File: book.rb [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+  attr_accessor :title, :author
+
+  attr_reader :rentals
+
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rentals = []
+  end
+end
```
Comment (Bondok6) [inline book.rb]: > Great job initialize `@rentals = [ ]` here 👍
- [ ] However, you should create a method for adding rentals and the book itself should be sent as a parameter to create the new rental achieving the 'has-many' association. (according to the requirements) 💯

> For Example 💡
```ruby
  def add_rental(person, date)
    Rental.new(date, self, person)
  end
```

File: person.rb [hunk 2]
```diff
@@ -10,6 +10,7 @@ def initialize(age, name = 'Unknown', parent_permission: true)
     @name = name
     @age = age
     @parent_permission = parent_permission
+    @rentals = []
   end
```
Comment (Bondok6) [inline person.rb:15]: - [ ] Exactly the same thing here ⬆, A method for adding rentals should be present in this class `Person` and the person itself should be sent as a parameter to create the new rental achieving the 'has-many' association 💯

> For Example 💡

```ruby
  def add_rental(book, date)
    Rental.new(date, book, self)
  end
```

**Review by Bondok6 — CHANGES_REQUESTED** [review]
## Hi @Bateyjosue 🙋‍♂️, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights
- No Linter errors ✅
- Correct GitHub flow ✅
- Your repo is professional ✅
- Your project looks great ✅
- 


## Required Changes ♻️
- [ ] _Check the comments under the review._ ⬇

## Optional suggestions

- [OPTIONAL] I noticed that the MIT Link is broken in your README file, so I suggest fixing it 
by changing the path from `[MIT](./MIT.md)` to `[MIT](./LICENSE)` 💯👍
> or just Rename the LICENSE file to MIT.md 😄

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by m05e5 — APPROVED** [review]
Hi @Bateyjosue , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
goodjob

## Highlights
✔️ your linters are all ok
✔️ Descriptive PR 

## Optional suggestions

 - [ optional ] Currently your category `check status` button is dispatching but its not displaying the message. Kindly fix it

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

_**m05e5**_

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**